### PR TITLE
WorkerTrainingState Test and Bugfix

### PR DESCRIPTION
**Pull request by sboshin**
When trying to backup and restoring from the same location on multiple workers, delete_backup has a bug where get_matching_files_v2 fails because another worker has already previously deleted the folder.

This has been fixed because of this [issue](https://github.com/tensorflow/tensorflow/issues/43789) still keeping the test there.

### CODE REVIEW

**Comment by gbaned** [comment]
@sboshin  Can you please resolve conflicts? Thanks!

**Comment by gbaned** [comment]
@sboshin Can you please resolve conflicts? Thanks!

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this PR still valid? Assigning the `stalled` label. Please comment to reassure me that this is still being worked on.

**Comment by gbaned** [comment]
@sboshin  Any update on this PR? Please. Thanks!

File: tensorflow/python/keras/distribute/worker_training_state_test.py [hunk 1]
```diff
@@ -17,46 +17,102 @@
...
+  train_ds, _ = multi_worker_testing_utils.mnist_synthetic_dataset(64, 2)
+  with strategy.scope():
+    model = multi_worker_testing_utils.get_mnist_model((28, 28, 1))
+
+  model._training_state = (worker_training_state.WorkerTrainingState(
+      model, backup_dir))
+  model._training_state.back_up(0)
+
+  len_backup = len(os.listdir(backup_dir))
+  assert len_backup > 0, "%d" % (len_backup)
```
Comment (Clyde-Xu) [inline tensorflow/python/keras/distribute/worker_training_state_test.py:88]: looks like it is possible for task0 to clean up the backup before others tasks reaching this line?

File: tensorflow/python/keras/distribute/worker_training_state_test.py [hunk 2]
```diff
@@ -17,46 +17,102 @@
...
+  def create_cluster_spec(self, num_workers):
+    worker_ports = [test_base.pick_unused_port() for _ in range(num_workers)]
+    cluster_dict = {}
+    if num_workers > 0:
+      cluster_dict['worker'] = ['localhost:%s' % port for port in worker_ports]
+    self._cluster_spec = cluster_dict
+
+  def testWorkerTrainingState(self):
+    num_workers = 3
+    self.create_cluster_spec(num_workers)
```
Comment (Clyde-Xu) [inline tensorflow/python/keras/distribute/worker_training_state_test.py:111]: optional: use test_base.create_cluster_spec

File: tensorflow/python/keras/distribute/worker_training_state.py [hunk 3]
```diff
@@ -21,6 +21,7 @@
 
 from tensorflow.python.framework import constant_op
 from tensorflow.python.framework import dtypes
+from tensorflow.python.framework.errors_impl import NotFoundError as _NotFoundError
```
Comment (Clyde-Xu) [inline tensorflow/python/keras/distribute/worker_training_state.py:24]: not used any more?

File: tensorflow/python/keras/distribute/worker_training_state.py [hunk 4]
```diff
@@ -43,15 +43,14 @@ class WorkerTrainingState(object):
...
-
   def __init__(self, model, checkpoint_dir):
     self._model = model
 
     # The epoch at which the checkpoint is saved. Used for fault-tolerance.
     # GPU device only has int64 dtype registered VarHandleOp.
     self._ckpt_saved_epoch = variables.Variable(
-        initial_value=constant_op.constant(
-            CKPT_SAVED_EPOCH_UNUSED_VALUE, dtype=dtypes.int64),
+        initial_value=constant_op.constant(CKPT_SAVED_EPOCH_UNUSED_VALUE,
```
Comment (rchao) [inline tensorflow/python/keras/distribute/worker_training_state.py:52]: Thanks for the PR! It looks like the changes in worker_training_state.py isn't necessary? Can we only keep the changes in the test?

**Review by Clyde-Xu — APPROVED** [review]

**Comment by sboshin** [comment]
Sorry it seems using multiprocessing is inheriting the eager context when running the test. I will switch to a subprocess test.

**Comment by rchao** [comment]
Thanks for the PR. Is it possible to not rely on multiprocessing module for unit tests? multiprocessing module in general does not work well with our testing infrastructure and unit tests without it will make it much smoother.

**Comment by mihaimaruseac** [comment]
Closing stale PR. Please reopen if still needed and wanting to work on it.
